wasm-runtime/polkadot/src/runtime/system.rs

```diff
@@ -73,13 +78,13 @@ pub mod internal {
 		// check transaction trie root represents the transactions.
 		let txs = block.transactions.iter().map(Slicable::to_vec).collect::<Vec<_>>();
 		let txs = txs.iter().map(Vec::as_slice).collect::<Vec<_>>();
-		let txs_root = enumerated_trie_root(&txs);
+		let txs_root = H256(enumerated_trie_root(&txs));
 		debug_assert_hash(&header.transaction_root, &txs_root);
 		assert!(header.transaction_root == txs_root, "Transaction trie root must be valid.");
 
 		// execute transactions
 		for tx in &block.transactions {
-			super::execute_transaction(tx);
+			super::execute_transaction(tx.clone());
```

Comment:
do we really need to clone?

Reply:
if we want to have the checked transaction type, yes. although we could further refactor into a `VerifiedBlock` type which can be constructed as it decodes the input block data.